### PR TITLE
Expose an API endpoint for transfer creation

### PR DESCRIPTION
**Pull request by LefterisJP**
Close https://github.com/raiden-network/raiden/issues/440

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -427,6 +427,66 @@ Possible Responses
 | 500 Server Error | Internal Raiden node error|
 +------------------+---------------------------+
 
+Transfers
+=========
+
+Initiating a Transfer
+---------------------
+
+You can create a new transfer by making a ``PUSH`` to the following endpoint along with a json payload containing
```
Comment (hackaugusto) [inline docs/api.rst]: PUSH -> POST

File: docs/api.rst [hunk 2]
```diff
@@ -427,6 +427,66 @@ Possible Responses
...
+the transfer details such as amount and identifier. Identifier is optional.
+
+``PUSH /api/<version>/transfers/<token_address>/<target_address>``
```
Comment (hackaugusto) [inline docs/api.rst]: PUSH -> POST

File: docs/api.rst [hunk 3]
```diff
@@ -427,6 +427,66 @@ Possible Responses
...
+
+Example Request
+^^^^^^^^^^^^^^^
+
+``PUSH /api/1/transfers/0x2a65aca4d5fc5b5c859090a6c34d164135398226/0x61c808d82a3ac53231750dadc13c777b59310bd9``
```
Comment (hackaugusto) [inline docs/api.rst]: PUSH -> PUT

File: docs/api.rst [hunk 4]
```diff
@@ -427,6 +427,66 @@ Possible Responses
...
+
+Initiating a Transfer
+---------------------
+
+You can create a new transfer by making a ``PUSH`` to the following endpoint along with a json payload containing
+the transfer details such as amount and identifier. Identifier is optional.
+
+``PUSH /api/<version>/transfers/<token_address>/<target_address>``
+
+Example Request
```
Comment (hackaugusto) [inline docs/api.rst]: What about adding a curl example?

```
curl -H "Content-Type: application/json" -X POST -d '{"amount": 200, "identifier": 42}' http://localhost:<port>/api/1/transfers....
```

File: docs/api.rst (hunk 1 above)
Comment (hackaugusto) [inline docs/api.rst]: I also think we should mention that this request will only return once the transfer succeeded or failed, and that the failure is determined (currently) by the lock's expiration (we need to implement initiator timeouts)

File: raiden/api/v1/resources.py [hunk 5]
```diff
@@ -121,3 +122,17 @@ def get(self, **kwargs):
...
+
+    post_schema = TransferSchema(
+        exclude=('initiator_address', 'target_address', 'token_address')
+    )
+
+    def __init__(self, **kwargs):
+        super(TransferToTargetResource, self).__init__(**kwargs)
+
+    @use_kwargs(post_schema, locations=('json',))
+    def post(self, **kwargs):
```
Comment (hackaugusto) [inline raiden/api/v1/resources.py]: We are abusing `**kwags` through out the API code, could you please use named arguments?

File: docs/api.rst (hunk 4 above)
Comment (LefterisJP) [inline docs/api.rst]: As I said in the other PR, we should add curl examples to all of the endpoints. It's a good idea but I would like to do it all in one swoop as 1 commit and it's not part of this PR.

File: docs/api.rst (hunk 3 above)
Comment (LefterisJP) [inline docs/api.rst]: you mean POST, right?

File: raiden/api/v1/resources.py (hunk 5 above)
Comment (LefterisJP) [inline raiden/api/v1/resources.py]: as discussed in the other PR I totallly agree. But I think it's only here and in the other PR I ever did it, so both will be fixed now.
